tests/integration/test_postgres.py

```diff
@@ -0,0 +1,150 @@
+    assert result == [
+        {
+            "id": pk,
+            "args": {"a": 1, "b": 2},
+            "status": "todo",
+            "targeted_object": "sher",
+            "task_type": "bob",
+        }
+    ]
+
+
+def test_launch_task_no_queue(db):
+    queue = "marsupilami"
+    with pytest.raises(exceptions.QueueNotFound):
+        postgres.launch_task(
+            queue=queue, name="bob", lock="sher", kwargs={"a": 1, "b": 2}
+        )
+
+
+def _check_next_task(it, end=False, **kwargs):
+    row = next(it)
+    if end:
+        assert row["id"] is None, f"iterator didn't end, yielded {row}"
+    else:
+        assert row.pop("id") is not None, f"no task found but expected {kwargs}"
+        row.pop("queue_id")
+        assert row == kwargs
+
+
+def test_get_tasks(db):
+    postgres.register_queue("queue_a")
+    postgres.register_queue("queue_b")
+    postgres.launch_task("queue_a", "task_1", "lock_1", {"a": "b"})
+    # We won't see this one because of the lock
+    postgres.launch_task("queue_a", "task_2", "lock_1", {"c": "d"})
+    postgres.launch_task("queue_a", "task_3", "lock_2", {"e": "f"})
+    # We won't see this one because of the queue
+    postgres.launch_task("queue_b", "task_4", "lock_3", {"g": "h"})
+
+    with postgres.get_dict_cursor(db) as cursor:
+        it = postgres.get_tasks(cursor, "queue_a")
+
+        _check_next_task(
+            it,
+            args={"a": "b"},
+            status="doing",
+            targeted_object="lock_1",
+            task_type="task_1",
+        )
+
+        _check_next_task(
+            it,
+            args={"e": "f"},
+            status="doing",
+            targeted_object="lock_2",
+            task_type="task_3",
+        )
+        _check_next_task(it, end=True)
+
+
+def test_finish_task(get_all, db):
+    postgres.register_queue("queue_a")
+    postgres.launch_task("queue_a", "task_1", "lock_1", {"a": "b"})
+    with postgres.get_dict_cursor(db) as cursor:
+        task = next(postgres.get_tasks(cursor, "queue_a"))
+
+        assert task["status"] == "doing"
+
+        postgres.finish_task(cursor=cursor, task_id=task["id"], status="done")
+
+    assert get_all("tasks", "status") == [{"status": "done"}]
+
+
+def test_register_queue(get_all):
+    pk = postgres.register_queue("marsupilami")
+
+    result = get_all("queues", "*")
+    assert result == [{"id": pk, "queue_name": "marsupilami"}]
+
+
+def test_register_queue_conflict(get_all):
+    postgres.register_queue("marsupilami")
+
+    pk = postgres.register_queue("marsupilami")
+
+    assert pk is None
+    result = get_all("queues", "queue_name")
+    assert result == [{"queue_name": "marsupilami"}]
+
+
+def test_listen_queue(db):
+    with db.cursor() as cursor:
+        postgres.listen_queue(curs=cursor, queue="bla")
+        db.commit()
+        cursor.execute("""select * from pg_listening_channels() """)
+        assert cursor.fetchone()[0] == "queue#bla"
```

Comment:
Ping @PierreDucroquet :tada:

Reply:
As you wish… mais c'est un peu jouer avec le feu, non ?
```
import uuid
queue = str(uuid.uuid4())
postgres.listen_queue(curs=cursor, queue=queue)
db.commit()
cursor.execute("""select * from pg_listening_channels() WHERE pg_listening_channels = %s""", (queue,))
assert cursor.fetchone()[0] == ("queue#%s" % queue)
```

=> t'es certain de pas te manger les pieds dans le tapis le jour où ton test unitaire évolue et où tu testes plusieurs choses en parallèle… Et ça coûte pas plus cher, non ?

Reply:
(let's keep it in :gb: , we're on an opensource project :) )

Play with fire, I'd say is a small overstatement. according to the doc, `pg_listening_channels` returns listeners for the current session. I'd tend to say that even if we were to run tests in parallel, on the same db, we'd need to use the same psycopg2 connection object to really start having problems.

That said, if you're more confortable that way, I have nothing against implementing that.

Reply:
(done)

Reply:
Sorry, forgot this one was public. As I said : "as you wish".
But "bla" is not a good english-looking name, if I may... :)

Reply:
You may. Foo bar baz is overrated sometimes. And I'm sure if we used other languages idle talk, especially non-ascii ones, we'd find interesting bug :D